tools/spectral/ipa/rulesets/functions/IPA106CreateMethodRequestBodyIsGetResponse.js

```diff
@@ -68,9 +81,19 @@ function checkViolationsAndReturnErrors(
     ];
   }
 
-  // Create filtered versions of schemas by removing properties with appropriate flags
-  const filteredCreateRequestSchema = removeWriteOnlyProperties(postMethodRequestContentPerMediaType);
-  const filteredGetResponseSchema = removeReadOnlyProperties(getMethodResponseContentPerMediaType);
+  const postRequestSchemaRef = getSchemaRef(postRequestContentPerMediaTypeUnresolved.schema);
+  const getResponseSchemaRef = getSchemaRef(getResponseContentPerMediaTypeUnresolved.schema);
+
+  if (postRequestSchemaRef && getResponseSchemaRef) {
+    const postRequestSchemaName = getSchemaNameFromRef(postRequestSchemaRef);
+    const getResponseSchemaName = getSchemaNameFromRef(getResponseSchemaRef);
+    if (postRequestSchemaName === getResponseSchemaName) {
+      return [];
+    }
+  }
```

Comment:
Why not compare all the refs instead. Separately this will not evaluate if both refs are empty. 

```suggestion
  if (postRequestSchemaRef && getResponseSchemaRef) {
    if (postRequestSchemaRef === getResponseSchemaRef) {
      return [];
    }
  }
```

Reply:
Applied on your behalf as this is minor correction. We can restore it if that is an issue.

Reply:
There are multiple ways to define components in openapi thus relying on last part of the reference is not reliable:

https://swagger.io/docs/specification/v3_0/components